spec/stud/interval_spec.rb

```diff
@@ -1,8 +1,19 @@
 require "stud/interval"
 require "spec_env" # from the top level spec/ directory
 
+module Stud
+  def reset!(target = Thread.current)
```

Comment:
> let's say we have a threadpool to reuse our threads and we stop a specific interval

I think I agree with the concern, but is this something we're likely to hit in Logstash? Will we want to stop a thread in a pool and reuse it later? Can you think of a use case where we'd want to stop things and then restart them --- specifically in a case where we wouldn't just use a new threadpool?